test-result-summary-client/src/utils/perf.js

```diff
@@ -1,22 +1,15 @@
 
-export const getParserProps = async () => {
-    let parserProps = await fetch( `/api/getParserProps`, {
+export const getBenchmarkMetricProps = async (benchmarkName, curMetric) => {
+    let parserProps = await fetch( `/api/getBenchmarkMetricProps?benchmarkName=${benchmarkName}`, {
             method: 'get'
         }
     );
     parserProps = await parserProps.json();
-    return parserProps;  
-}
-
-export function getMetricProps(parserProps,benchmarkName,curMetric) {
-    let benchmarkMetric = parserProps['benchmarkMetric'];
+    let metrics = parserProps["metrics"];
     try {
-        for(let key of Object.keys(benchmarkMetric)){
-            if(benchmarkName.includes(key)){
-                return benchmarkMetric[key]["metrics"][curMetric]
-            }
-        }
+        return metrics[curMetric]
     } catch (benchmarkInfoErr) {
         return "";
     }
+    
```

Comment:
With this logic, we call the API per `curMetric ` for the same benchmarkName. We should only get info for the same benchmarkName once and set into the state.